src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -60,8 +60,17 @@ public static void Resize<T>([NotNull] ref T[]? array, int newSize)
 
             if (larray.Length != newSize)
             {
+                // Due to array variance, it's possible that the incoming array is
+                // actually of type U[], where U:T; or that an int[] <-> uint[] or
+                // similar cast has occurred. In any case, since it's always legal
+                // to reinterpret U as T in this scenario (but not necessarily the
+                // other way around), we can use Buffer.Memmove here.
+
                 T[] newArray = new T[newSize];
-                Copy(larray, 0, newArray, 0, larray.Length > newSize ? newSize : larray.Length);
+                Buffer.Memmove(
```

Comment:
```suggestion
                Buffer.Memmove<T>(
```
Just for clarity, to highlight that T is actually taken into account by the Memmove call.